src/biguint.rs

```diff
@@ -639,6 +656,21 @@ impl AddAssign<u64> for BigUint {
             }
         }
     }
+
+    #[cfg(u64_digit)]
+    #[inline]
+    fn add_assign(&mut self, other: u64) {
+        if other != 0 {
+            if self.data.len() == 0 {
+                self.data.push(0);
+            }
+
+            let carry = __add2(&mut self.data, &[other as BigDigit]);
```

Comment:
The implementation of `__add2` has the comment `// Only for the Add impl:` above it. Since it no longer is only for add, maybe it can get some (short) documentation. It is still a private function of course.

It seems like `__add2` does something like "add assign ignoring carry, then return the carry". It might even make sense to rename `__add2`, but that it probably out of scope here.

Reply:
I think `__add2` predates `AddAssign`, so it was formerly just for `Add`. Then when we implemented `AddAssign`, it was convenient to forward `Add` to that.

I added some documentation to it.